functions/geometry_vi.zsh

```diff
@@ -0,0 +1,25 @@
+# A geometry plugin to display the ZLE vi-mode state.
+
+update_rprompt() {
+    RPROMPT=${${RPROMPT/\[NORMAL\]/}/\[INSERT\]/}
+    RPROMPT="[$1]$RPROMPT"
+}
+
+update_prompt() {
+    PROMPT=${${PROMPT/\[NORMAL\]/}/\[INSERT\]/}
+    PROMPT="[$1]$PROMPT"
+}
+
+function zle-keymap-select {
```

Comment:
can we prefix this function with `geometry-` or `zle-geometry-vi-` so its clear where its coming from?

Reply:
`zle-keymap-select` is a special  widget so we can't have a user defined widget if we want to receive a callback every time the keymap chages, but we can definitely rename the callback function using the `zle -N zle-keymap-select geometry-vi-keymap-select` syntax.

Reply:
That would be great to rename the callback function and bind as you described.